bench/app.py

```diff
@@ -98,23 +112,21 @@ def _setup_details_from_mounted_disk(self):
 	def _setup_details_from_name_tag(self):
 		self.org, self.repo, self.tag = fetch_details_from_tag(self.name)
 
+		if self.tag and self.branch:
+			click.secho(
+				"Ignoring the branch provided with `--branch` option",
+				fg="bright_yellow"
+			)
+		elif not self.tag and self.branch:
+			self.tag = self.branch
```

Comment:
@Alchez i've added this logic for picking up the branch when partial url is provided - so as we can also provide branch with `--branch` flag, i'm ignoring the that value when the partial url has branch provided in itself (i.e `org/repo@branch`) - otherwise when the partial url doesn't have the branch provided but is provided by `--branch` flag, we take the flag's value

Reply:
@phot0n, yeah that makes sense. I would also suggest changing any function names that have `name-tag` to `partial-url`, to keep it consistent

Reply:
done :)
Please do have a look ...

Reply:
Yeah, looks good now